### PR TITLE
Apply proxy settings for non-secure sockets.

### PR DESCRIPTION
**Pull request by jonas-lundqvist**

### CODE REVIEW

**Comment by jonas-lundqvist** [comment]
This should be mentioned in #980 
And the same thing should also be applied to (unsecure) WebDAV connections.

**Comment by philipwhiuk** [comment]
For WebDAV insecure proxy support we'll need to register a scheme for HTTP and replace the socket factory.with one that does proxy support. Currently we just leave HTTP alone as we only have to fiddle with the HTTPS one. 

I do wonder what the usage for insecure WebDAV over a proxy is.

File: k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java [hunk 1]
```diff
@@ -121,8 +121,7 @@
 
 
     private Context context;
-    private ProxySettings proxySettings;
-
+    private ProxySettings proxySettings = new ProxySettings(false, "", 0);
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java]: What's that good for? The field is initialized in the constructor.

File: k9mail-library/src/main/java/com/fsck/k9/mail/ssl/TrustedSocketFactory.java [hunk 2]
```diff
@@ -10,4 +11,6 @@
 public interface TrustedSocketFactory {
     Socket createSocket(Socket socket, String host, int port, String clientCertificateAlias)
             throws NoSuchAlgorithmException, KeyManagementException, MessagingException, IOException;
+
+    ProxySettings getProxySettings();
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/ssl/TrustedSocketFactory.java]: This doesn't belong in `TrustedSocketFactory`.

We probably want a general socket factory that knows how to create trusted and regular sockets. Then all the proxy code can be moved there.

**Comment by cketti** [comment]
I decided to do this a bit differently. We'll pass a socket factory to the remote stores that will deal with the proxy settings.

**Pull request closed without merging**